R/process_data.R

```diff
@@ -133,6 +141,40 @@ remove_companies_with_missing_exposures <- function(data,
   return(data_filtered)
 }
 
+#' Rows that have no information on the exposure before the end of the
+#' production forecast, should get a zero exposure value. This allows keeping
+#' the information for cases where production is only being built out toward the
+#' end of the forecast period.
+#'
+#' @inheritParams calculate_annual_profits
+#' @inheritParams report_company_drops
+#' @param data tibble containing filtered PACTA results
+#'
+#' @return A tibble of data without rows with no exposure info
+#' @noRd
+set_initial_plan_carsten_missings_to_zero <- function(data,
+                                                      start_year,
+                                                      time_horizon) {
+  companies_missing_exposure_value <- data %>%
+    dplyr::filter(.data$year < .env$start_year + .env$time_horizon) %>%
+    dplyr::filter(is.na(.data$plan_carsten)) %>%
+    dplyr::mutate(plan_carsten = 0)
+
+  data_replaced <- data %>%
+    dplyr::filter(
```

Comment:
suggestion: how about instead of the filtering and rowbinding approach simply using mutate(if_else())